core/chains/evm/txmgr/confirmer_test.go

```diff
@@ -3122,6 +3123,7 @@ func TestEthConfirmer_ResumePendingRuns(t *testing.T) {
 		case <-time.After(time.Second):
 			t.Fatal("no value received")
 		}
+		wg.Wait()
```

Comment:
Making sure the unit test doesn't exit (t.context cancelled) before the [goroutine](https://github.com/smartcontractkit/chainlink/blob/a234e14ebd266269b4d5893b0d2aeeb01bc58a70/core/chains/evm/txmgr/confirmer_test.go#L3155) returns. Right now we are getting context cancelled [error](https://chainlink-core.slack.com/archives/C05JDS9S64S/p1725292620722479) at [line](https://github.com/smartcontractkit/chainlink/blob/a234e14ebd266269b4d5893b0d2aeeb01bc58a70/core/chains/evm/txmgr/evm_tx_store.go#L1084)

Reply:
This [error](https://chainlink-core.slack.com/archives/C05JDS9S64S/p1725292620722479) can be reproduced by putting a breakpoint at [line](https://github.com/smartcontractkit/chainlink/blob/a234e14ebd266269b4d5893b0d2aeeb01bc58a70/core/chains/evm/txmgr/evm_tx_store.go#L1084)

Reply:
> Making sure the unit test doesn't exit (t.context cancelled) before the goroutine returns.

It does not achieve this. The old code enforced this strictly and minimally. The new code can return before reaching this line.

Reply:
Can you explain the leak ?

Reply:
There is no guarantee that this line with `Wait()` will execute. The test can return earlier. That's why it is essential to register cleanup conditions along with spawning the goroutine, just like the old code did. No matter when you return, the test will not complete until the goroutine yields.

Reply:
Do you know for sure the registered Cleanup function is called before context cancelled ?

Reply:
> Do you know for sure the registered Cleanup function is called before context cancelled ?

No. What do you mean? Nothing should be making any assumptions about that. It would be racey to do so. The context can either be cancelled via the parent that is tied to the test deadline, or via the stopChan which is trigged by close during cleanup.